test/components/Home.vue

```diff
@@ -23,34 +59,99 @@
  */
 'use strict';
 
-import {StatsService} from 'bedrock-web-stats';
+// FIXME: chartjs is loaded as a global in index.js as a hack
+import {ChartController} from 'bedrock-web-stats';
+import {BrGaugeChart, BrTimeSeriesChart} from 'bedrock-vue-stats';
+import units from './units';
 
 export default {
   name: 'Home',
-  components: {},
+  components: {BrGaugeChart, BrTimeSeriesChart},
   data() {
     return {
-      reports: []
+      charts: {last: {}},
+      statsService: null,
+      cpu: null,
+      ram: null,
+      disk: null,
+      cpuseries: null,
+      ramseries: null,
+      diskseries: null
+
     };
   },
-  beforeCreate() {
-    this._statsService = new StatsService();
+  computed: {
+    loading() {
+      if(!this.cpu) {
+        return true;
+      }
+      return this.cpu.loading;
+    }
+  },
+  mounted() {
+    this.$set(this, 'cpu', new ChartController(
```

Comment:
Is this the intended interaction with the API?

- Why are we using `$set()` on variables that are created in `data()`?
- For `format` it seems to me that `prefix` is quite misleading. I believe
  this function provides the location of the data in an object for use of
  display if I am not mistaken. How can we name this better to fit its
  true purpose?

Reply:
Also, don't know how Vue reactivity works with the instantiation of a class. Do the proper getters and setters work?

Reply:
@gannan08 $set guarantees an update. Vue's reactivity actually will not update if you mutate the data and while I believe you are correct that replacing null with a class will be noticed and cause an update I would prefer that we adopt using this.$set in all cases (I have worked on apps where people pushed to an array which resulted in vue not updating etc.) using $set is just good form IMHO. Every Vue project I have worked on I have observed cases where people do not understand Vue's reactivity so I use $set in all cases in order to be sure.

I agree on prefix is a bad name maybe one of these options:
- getter
- prepare or preparer
- formatter

I think prepare is good actually.

Reply:
as for the getters and setters I guess you mean the class? does seem to be working not having any issues. Good concern though.